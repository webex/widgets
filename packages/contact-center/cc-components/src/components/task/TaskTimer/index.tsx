import React, {useEffect, useState, useRef} from 'react';

interface TaskTimerProps {
  startTimeStamp?: number; // in milliseconds
  countdown?: boolean;
  ronaTimeout?: number; // in seconds
}

const workerScript = `
  const formatDuration = (seconds) => {
    seconds = Math.max(0, seconds);
    const h = Math.floor(seconds / 3600).toString().padStart(2, '0');
    const m = Math.floor((seconds % 3600) / 60).toString().padStart(2, '0');
    const s = Math.floor(seconds % 60).toString().padStart(2, '0');
    return seconds < 3600 ? m + ':' + s : h + ':' + m + ':' + s;
  }

  const timers = {};

  self.onmessage = function(e) {
    const { type, name, start, countdown, ronaTimeout } = e.data;

    if (type === 'start') {
      if (timers[name]) {
        return; // Timer with this name is already running
      }

      let running = true;

      const updateTimer = () => {
        if (!running) {
          return;
        }

        const now = Math.floor(Date.now() / 1000); // current time in seconds
        let diff;

        if (countdown && ronaTimeout !== undefined) {
          diff = start + ronaTimeout - now;
          if (diff <= 0) {
            self.postMessage({ name, time: '00:00' });
            running = false;
            delete timers[name];
            return;
          }
        } else {
          diff = now - start;
        }

        const formattedTime = formatDuration(diff);
        self.postMessage({ name, time: formattedTime });
        timers[name] = setTimeout(updateTimer, 1000);
      };

      updateTimer();
    } else if (type === 'stop') {
      if (timers[name]) {
        clearTimeout(timers[name]);
        delete timers[name];
      }
    }
  };
`;

const createWorker = () => {
  return new Worker(URL.createObjectURL(new Blob([workerScript], {type: 'application/javascript'})));
};

const TaskTimer: React.FC<TaskTimerProps> = ({startTimeStamp, countdown = false, ronaTimeout}) => {
  const [duration, setDuration] = useState('00:00');
  const workerRef = useRef<Worker | null>(null);
  const timerName = useRef(`timer-${Date.now()}`); // Unique name for the timer

  useEffect(() => {
    const now = Math.floor(Date.now() / 1000);
    const start = startTimeStamp ? Math.floor(startTimeStamp / 1000) : now;

    if (!workerRef.current) {
      workerRef.current = createWorker();
    }

    workerRef.current.postMessage({type: 'start', name: timerName.current, start, countdown, ronaTimeout});

    const handleMessage = (e: MessageEvent) => {
      if (e.data.name === timerName.current) {
        setDuration(e.data.time);
      }
    };

    workerRef.current.addEventListener('message', handleMessage);

    return () => {
      workerRef.current.postMessage({type: 'stop', name: timerName.current});
      workerRef.current?.removeEventListener('message', handleMessage);
    };
  }, [countdown, ronaTimeout, startTimeStamp]);

  return (
    <time dateTime={duration} className="task-text task-text--secondary">
      {duration}
    </time>
  );
};

export default TaskTimer;
