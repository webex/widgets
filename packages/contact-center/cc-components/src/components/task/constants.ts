// Shared text constants for task components
export const UNTIL_AUTO_WRAPUP = 'Until auto wrap-up';
export const CANCEL = 'Cancel';
export const RESUME_CALL = 'Resume the call';
export const HOLD_CALL = 'Hold the call';
export const CONSULT_AGENT = 'Consult with another agent';
export const TRANSFER = 'Transfer';
export const PAUSE_RECORDING = 'Pause Recording';
export const RESUME_RECORDING = 'Resume Recording';
export const END = 'End';
export const WRAP_UP = 'Wrap up';
export const WRAP_UP_INTERACTION = 'Wrap up interaction';
export const WRAP_UP_REASON = 'Wrap-up reason';
export const SELECT = 'Select';
export const SUBMIT_WRAP_UP = 'Submit & Wrap up';
export const MUTE_CALL = 'Mute';
export const UNMUTE_CALL = 'Unmute';
export const DIAL_NUMBER = 'DialNumber';
export const AGENTS = 'Agents';
export const QUEUES = 'Queues';
// CallControlCAD constants
export const NO_CUSTOMER_NAME = 'No Customer Name';
export const NO_CALLER_ID = 'No Caller ID';
export const NO_PHONE_NUMBER = 'No Phone Number';
export const NO_TEAM_NAME = 'No Team Name';
export const NO_RONA = 'No RONA';
export const ON_HOLD = 'On hold -';
export const QUEUE = 'Queue:';
export const PHONE_NUMBER = 'Phone Number:';
export const CUSTOMER_NAME = 'Customer Name';
export const RONA = 'RONA:';
