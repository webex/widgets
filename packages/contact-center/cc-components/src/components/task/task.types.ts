import {ILogger, ITask, IContactCenter, WrapupCodes} from '@webex/cc-store';

/**
 * Interface representing the TaskProps of a user.
 */
export interface TaskProps {
  /**
   * currentTask of the agent.
   */
  currentTask: ITask;

  /**
   * Incoming task on the incoming task widget
   */
  incomingTask: ITask;

  /**
   * CC SDK Instance.
   */
  cc: IContactCenter;

  /**
   * Handler for task accepted
   */
  onAccepted?: () => void;

  /**
   * Handler for task declined
   */
  onDeclined?: () => void;

  /**
   * Handler for task accepted in TaskList
   */
  onTaskAccepted?: (task: ITask) => void;

  /**
   * Handler for task declined in TaskList
   */
  onTaskDeclined?: (task: ITask) => void;

  /**
   * accept incoming task action
   */
  accept: () => void;

  /**
   * decline incoming task action
   */
  decline: () => void;

  /**
   * accept task from task list
   */
  acceptTask: (task: ITask) => void;

  /**
   * decline task from tasklist
   */
  declineTask: (task: ITask) => void;

  /**
   * Flag to determine if the user is logged in with a browser option
   */
  isBrowser: boolean;

  /**
   * Flag to determine if the task is answered
   */
  isAnswered: boolean;

  /**
   * Flag to determine if the task is ended
   */
  isEnded: boolean;

  /**
   * Selected login option
   */
  deviceType: string;

  /**
   * List of tasks
   */
  taskList: ITask[];

  /**
   * The logger instance from SDK
   */
  logger: ILogger;
}

export type UseTaskProps = Pick<TaskProps, 'incomingTask' | 'onAccepted' | 'onDeclined' | 'deviceType' | 'logger'>;
export type UseTaskListProps = Pick<
  TaskProps,
  'cc' | 'taskList' | 'deviceType' | 'onTaskAccepted' | 'onTaskDeclined' | 'logger'
>;
export type IncomingTaskPresentationalProps = Pick<TaskProps, 'incomingTask' | 'isBrowser' | 'accept' | 'decline'>;
export type IncomingTaskProps = Pick<TaskProps, 'onAccepted' | 'onDeclined'>;
export type TaskListProps = Pick<TaskProps, 'onTaskAccepted' | 'onTaskDeclined'>;

export type TaskListPresentationalProps = Pick<
  TaskProps,
  'currentTask' | 'taskList' | 'isBrowser' | 'acceptTask' | 'declineTask'
>;

/**
 * Interface representing the properties for control actions on a task.
 */
export interface ControlProps {
  /**
   * Audio reference
   */
  audioRef: React.RefObject<HTMLAudioElement>;
  /**
   * The current task being handled.
   */
  currentTask: ITask;

  /**
   * Function to handle hold/resume actions.
   */
  onHoldResume?: () => void;

  /**
   * Function to handle ending the task.
   */
  onEnd?: () => void;

  /**
   * Function to handle wrapping up the task.
   */
  onWrapUp?: ({task, wrapUpReason}: {task: ITask; wrapUpReason: string}) => void;

  /**
   * Logger instance for logging purposes.
   */
  logger: ILogger;

  /**
   * Array of wrap-up codes.
   * TODO: Expose this type from SDK.
   */
  wrapupCodes: WrapupCodes[];

  /**
   * Indicates if wrap-up is required.
   */
  wrapupRequired: boolean;

  /**
   * Function to handle hold/resume actions with a boolean parameter.
   * @param hold - Boolean indicating whether to hold (true) or resume (false).
   */
  toggleHold: (hold: boolean) => void;

  /**
   * Function to handle pause/resume recording actions with a boolean parameter.
   * @param pause - Boolean indicating whether to pause (true) or resume (false) recording.
   */
  toggleRecording: (pause: boolean) => void;

  /**
   * Function to handle ending the call.
   */
  endCall: () => void;

  /**
   * Function to handle wrapping up the call with a reason and ID.
   * @param wrapupReason - The reason for wrapping up the call.
   * @param wrapupId - The ID associated with the wrap-up reason.
   */
  wrapupCall: (wrapupReason: string, wrapupId: string) => void;

  /**
   * Selected login option
   */
  deviceType: string;

  /**
   * Flag to determine if the task is held
   */
  isHeld: boolean;

  /**
   * Function to set the held status of the task.
   * @param isHeld - Boolean indicating whether the task is held.
   */
  setIsHeld: (isHeld: boolean) => void;
}

export type CallControlProps = Pick<ControlProps, 'onHoldResume' | 'onEnd' | 'onWrapUp'>;

export type CallControlPresentationalProps = Pick<
  ControlProps,
  | 'currentTask'
  | 'audioRef'
  | 'wrapupCodes'
  | 'wrapupRequired'
  | 'toggleHold'
  | 'toggleRecording'
  | 'endCall'
  | 'wrapupCall'
  | 'isHeld'
  | 'setIsHeld'
>;

export type useCallControlProps = Pick<
  ControlProps,
  'currentTask' | 'onHoldResume' | 'onEnd' | 'onWrapUp' | 'logger' | 'deviceType'
>;
