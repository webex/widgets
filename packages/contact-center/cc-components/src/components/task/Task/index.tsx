import React from 'react';
import {ButtonPill, ListItemBase, ListItemBaseSection, Text} from '@momentum-ui/react-collaboration';
import {Avatar} from '@momentum-design/components/dist/react';
import {PressEvent} from '@react-types/shared';
import TaskTimer from '../TaskTimer';
import './styles.scss';

interface TaskProps {
  interactionId?: string;
  title?: string;
  state?: string;
  startTimeStamp?: number;
  ronaTimeout?: number;
  selected?: boolean;
  isIncomingTask?: boolean;
  queue?: string;
  acceptTask?: (e: PressEvent) => void;
  declineTask?: (e: PressEvent) => void;
  onTaskSelect?: (e: PressEvent) => void;
  acceptText?: string;
  declineText?: string;
  disableAccept?: boolean;
  styles?: string;
}

const Task: React.FC<TaskProps> = ({
  title,
  state,
  startTimeStamp,
  ronaTimeout,
  selected = false,
  styles,
  isIncomingTask = false,
  queue,
  acceptTask,
  declineTask,
  interactionId,
  onTaskSelect,
  acceptText,
  disableAccept = false,
  declineText,
}) => {
  const capitalizeFirstWord = (str: string) => {
    return str.replace(/^\s*(\w)/, (match, firstLetter) => firstLetter.toUpperCase());
  };

  return (
    <ListItemBase
      className={`task-list-item ${selected ? 'task-list-item--selected' : ''} ${styles}`}
      onPress={onTaskSelect ? onTaskSelect : undefined}
      id={interactionId}
    >
      <ListItemBaseSection position="start">
        <Avatar icon-name="handset-filled" />
      </ListItemBaseSection>

      <ListItemBaseSection position="fill">
        <section className="task-details">
          {title && (
            <Text tagName="span" type={selected ? 'body-large-bold' : 'body-large-medium'}>
              {title}
            </Text>
          )}

          {state && !isIncomingTask && (
            <Text tagName="span" type="body-midsize-regular" className="task-text">
              {capitalizeFirstWord(state)}
            </Text>
          )}

          {queue && isIncomingTask && (
            <Text tagName="span" type="body-midsize-regular" className="task-text">
              {capitalizeFirstWord(queue)}
            </Text>
          )}

          {/* Handle Time should render if it's an incoming call without ronaTimeout OR if it's not an incoming call */}
          {(isIncomingTask && !ronaTimeout) || !isIncomingTask
            ? startTimeStamp && (
                <Text tagName="span" type="body-midsize-regular" className="task-text">
                  Handle Time: {'  '}
                  <TaskTimer startTimeStamp={startTimeStamp} />
                </Text>
              )
            : null}

          {/* Time Left should render if it's an incoming call with ronaTimeout */}
          {isIncomingTask && ronaTimeout && (
            <Text tagName="span" type="body-midsize-regular" className="task-text">
              Time Left: {'  '}
              <TaskTimer countdown={true} ronaTimeout={ronaTimeout} />
            </Text>
          )}
        </section>
      </ListItemBaseSection>

      <ListItemBaseSection position="end">
        <div className="task-button-container">
          {acceptText ? (
            <ButtonPill onPress={acceptTask} color="join" disabled={disableAccept}>
              {acceptText}
            </ButtonPill>
          ) : null}
          {declineText ? (
            <ButtonPill onPress={declineTask} color="cancel">
              {declineText}
            </ButtonPill>
          ) : null}
        </div>
      </ListItemBaseSection>
    </ListItemBase>
  );
};

export default Task;
