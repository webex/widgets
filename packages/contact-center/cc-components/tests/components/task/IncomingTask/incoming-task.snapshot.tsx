import React from 'react';
import {render} from '@testing-library/react';
import IncomingTaskComponent from '../../../../src/components/task/IncomingTask/incoming-task';
import {MEDIA_CHANNEL} from '../../../../src/components/task/task.types';

// Mock TaskTimer component
jest.mock('../../../../src/components/task/TaskTimer', () => {
  return {
    __esModule: true,
    default: () => <div data-testid="mock-timer">Timer</div>,
  };
});

describe('IncomingTaskComponent', () => {
  // Mock logger
  const mockLogger = {
    info: jest.fn(),
    warn: jest.fn(),
    error: jest.fn(),
    debug: jest.fn(),
  };

  // Helper function to create mock tasks
  const createMockTask = (overrides = {}) => ({
    data: {
      interactionId: 'test-interaction-123',
      wrapUpRequired: false,
      interaction: {
        callAssociatedDetails: {
          ani: '1234567890',
          customerName: 'John Doe',
          virtualTeamName: 'Support Team',
          ronaTimeout: '30',
        },
        createdTimestamp: 1641234567890,
        mediaType: MEDIA_CHANNEL.TELEPHONY,
        mediaChannel: 'voice',
      },
      ...overrides,
    },
  });

  const defaultProps = {
    accept: jest.fn(),
    reject: jest.fn(),
    logger: mockLogger,
  };

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('renders empty component when incomingTask is null', () => {
    const props = {
      ...defaultProps,
      incomingTask: null,
      isBrowser: true,
    };

    const {container} = render(<IncomingTaskComponent {...props} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders incoming call and handles accept/decline for browser telephony task', () => {
    const mockTask = createMockTask();
    const props = {
      ...defaultProps,
      incomingTask: mockTask,
      isBrowser: true,
    };

    const {container} = render(<IncomingTaskComponent {...props} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('shows correct button text for non-browser telephony task', () => {
    const mockTask = createMockTask();
    const props = {
      ...defaultProps,
      incomingTask: mockTask,
      isBrowser: false,
    };

    const {container} = render(<IncomingTaskComponent {...props} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('handles social media task correctly', () => {
    const mockTask = createMockTask({
      interaction: {
        callAssociatedDetails: {
          ani: '1234567890',
          customerName: 'Alice Johnson',
          virtualTeamName: 'Social Media Team',
          ronaTimeout: '60',
        },
        createdTimestamp: 1641234567890,
        mediaType: MEDIA_CHANNEL.SOCIAL,
        mediaChannel: 'facebook',
      },
    });
    const props = {
      ...defaultProps,
      incomingTask: mockTask,
      isBrowser: true,
    };

    const {container} = render(<IncomingTaskComponent {...props} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders browser telephony task with wrap up required', () => {
    const mockTask = createMockTask({
      wrapUpRequired: true,
    });
    const props = {
      ...defaultProps,
      incomingTask: mockTask,
      isBrowser: true,
    };

    const {container} = render(<IncomingTaskComponent {...props} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders social media task with wrap up required', () => {
    const mockTask = createMockTask({
      wrapUpRequired: true,
      interaction: {
        callAssociatedDetails: {
          ani: '9876543210',
          customerName: 'Bob Wilson',
          virtualTeamName: 'Social Support',
        },
        createdTimestamp: 1641234567890,
        mediaType: MEDIA_CHANNEL.SOCIAL,
        mediaChannel: 'twitter',
      },
    });
    const props = {
      ...defaultProps,
      incomingTask: mockTask,
      isBrowser: true,
    };

    const {container} = render(<IncomingTaskComponent {...props} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders chat task without wrap up', () => {
    const mockTask = createMockTask({
      interaction: {
        callAssociatedDetails: {
          ani: 'chat-user-123',
          customerName: 'Chat Customer',
          virtualTeamName: 'Chat Support',
        },
        createdTimestamp: 1641234567890,
        mediaType: MEDIA_CHANNEL.CHAT,
        mediaChannel: 'webchat',
      },
    });
    const props = {
      ...defaultProps,
      incomingTask: mockTask,
      isBrowser: true,
    };

    const {container} = render(<IncomingTaskComponent {...props} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders email task without wrap up', () => {
    const mockTask = createMockTask({
      interaction: {
        callAssociatedDetails: {
          ani: 'customer@example.com',
          customerName: 'Email Customer',
          virtualTeamName: 'Email Support',
        },
        createdTimestamp: 1641234567890,
        mediaType: MEDIA_CHANNEL.EMAIL,
        mediaChannel: 'email',
      },
    });
    const props = {
      ...defaultProps,
      incomingTask: mockTask,
      isBrowser: true,
    };

    const {container} = render(<IncomingTaskComponent {...props} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders task with missing call association details', () => {
    const mockTask = createMockTask({
      interaction: {
        callAssociatedDetails: undefined,
        createdTimestamp: 1641234567890,
        mediaType: MEDIA_CHANNEL.TELEPHONY,
        mediaChannel: 'voice',
      },
    });
    const props = {
      ...defaultProps,
      incomingTask: mockTask,
      isBrowser: true,
    };

    const {container} = render(<IncomingTaskComponent {...props} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders task with special characters in ANI', () => {
    const mockTask = createMockTask({
      interaction: {
        callAssociatedDetails: {
          ani: '+1 (555) 123-4567 ext. 123',
          customerName: 'Special Char Customer',
          virtualTeamName: 'Special Support & Services',
        },
        createdTimestamp: 1641234567890,
        mediaType: MEDIA_CHANNEL.TELEPHONY,
        mediaChannel: 'voice',
      },
    });
    const props = {
      ...defaultProps,
      incomingTask: mockTask,
      isBrowser: true,
    };

    const {container} = render(<IncomingTaskComponent {...props} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
