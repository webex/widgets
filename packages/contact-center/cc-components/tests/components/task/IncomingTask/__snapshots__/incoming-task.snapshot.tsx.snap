// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IncomingTaskComponent handles social media task correctly 1`] = `
<li
  class="task-list-item  task-list-hover md-list-item-base-wrapper"
  data-allow-text-select="false"
  data-disabled="false"
  data-interactive="true"
  data-padded="false"
  data-shape="rectangle"
  data-size="40"
  id="test-interaction-123"
  role="listitem"
  tabindex="0"
>
  <div
    data-position="start"
  >
    <div
      class="brand-visual-background"
    >
      <mdc-brandvisual
        class="facebook"
      />
    </div>
  </div>
  <div
    data-position="fill"
  >
    <section
      class="task-details"
    >
      <mdc-text
        class="md-text-wrapper incoming-digital-task-title"
        id="tooltip-trigger-test-interaction-123"
      >
        Alice Johnson
      </mdc-text>
      <mdc-tooltip
        class="task-tooltip"
        tooltip-type="description"
      >
        Alice Johnson
      </mdc-tooltip>
      <mdc-text
        class="md-text-wrapper task-text"
      >
        Social Media Team
      </mdc-text>
      <mdc-text
        class="md-text-wrapper task-text"
      >
        Time Left: 
          
        <div
          data-testid="mock-timer"
        >
          Timer
        </div>
      </mdc-text>
    </section>
  </div>
  <div
    data-position="end"
  >
    <div
      class="task-button-container"
    >
      <button
        class="md-button-pill-wrapper md-button-simple-wrapper"
        data-color="join"
        data-disabled="false"
        data-disabled-outline="false"
        data-ghost="false"
        data-grown="false"
        data-inverted="false"
        data-outline="false"
        data-shallow-disabled="false"
        data-size="40"
        data-testid="task:accept-button"
        tabindex="-1"
        type="button"
      >
        <span>
          Accept
        </span>
      </button>
    </div>
  </div>
</li>
`;

exports[`IncomingTaskComponent renders browser telephony task with wrap up required 1`] = `
<li
  class="task-list-item  task-list-hover md-list-item-base-wrapper"
  data-allow-text-select="false"
  data-disabled="false"
  data-interactive="true"
  data-padded="false"
  data-shape="rectangle"
  data-size="40"
  id="test-interaction-123"
  role="listitem"
  tabindex="0"
>
  <div
    data-position="start"
  >
    <mdc-avatar
      class="telephony"
      icon-name="handset-filled"
    />
  </div>
  <div
    data-position="fill"
  >
    <section
      class="task-details"
    >
      <mdc-text
        class="md-text-wrapper task-title"
      >
        1234567890
      </mdc-text>
      <mdc-text
        class="md-text-wrapper task-text"
      >
        Support Team
      </mdc-text>
      <mdc-text
        class="md-text-wrapper task-text"
      >
        Time Left: 
          
        <div
          data-testid="mock-timer"
        >
          Timer
        </div>
      </mdc-text>
    </section>
  </div>
  <div
    data-position="end"
  >
    <div
      class="task-button-container"
    />
  </div>
</li>
`;

exports[`IncomingTaskComponent renders chat task without wrap up 1`] = `
<li
  class="task-list-item  task-list-hover md-list-item-base-wrapper"
  data-allow-text-select="false"
  data-disabled="false"
  data-interactive="true"
  data-padded="false"
  data-shape="rectangle"
  data-size="40"
  id="test-interaction-123"
  role="listitem"
  tabindex="0"
>
  <div
    data-position="start"
  >
    <mdc-avatar
      class="chat"
      icon-name="chat-filled"
    />
  </div>
  <div
    data-position="fill"
  >
    <section
      class="task-details"
    >
      <mdc-text
        class="md-text-wrapper incoming-digital-task-title"
        id="tooltip-trigger-test-interaction-123"
      >
        chat-user-123
      </mdc-text>
      <mdc-tooltip
        class="task-tooltip"
        tooltip-type="description"
      >
        chat-user-123
      </mdc-tooltip>
      <mdc-text
        class="md-text-wrapper task-text"
      >
        Chat Support
      </mdc-text>
      <mdc-text
        class="md-text-wrapper task-text"
      >
        Handle Time: 
          
        <div
          data-testid="mock-timer"
        >
          Timer
        </div>
      </mdc-text>
    </section>
  </div>
  <div
    data-position="end"
  >
    <div
      class="task-button-container"
    >
      <button
        class="md-button-pill-wrapper md-button-simple-wrapper"
        data-color="join"
        data-disabled="false"
        data-disabled-outline="false"
        data-ghost="false"
        data-grown="false"
        data-inverted="false"
        data-outline="false"
        data-shallow-disabled="false"
        data-size="40"
        data-testid="task:accept-button"
        tabindex="-1"
        type="button"
      >
        <span>
          Accept
        </span>
      </button>
    </div>
  </div>
</li>
`;

exports[`IncomingTaskComponent renders email task without wrap up 1`] = `
<li
  class="task-list-item  task-list-hover md-list-item-base-wrapper"
  data-allow-text-select="false"
  data-disabled="false"
  data-interactive="true"
  data-padded="false"
  data-shape="rectangle"
  data-size="40"
  id="test-interaction-123"
  role="listitem"
  tabindex="0"
>
  <div
    data-position="start"
  >
    <mdc-avatar
      class="email"
      icon-name="email-filled"
    />
  </div>
  <div
    data-position="fill"
  >
    <section
      class="task-details"
    >
      <mdc-text
        class="md-text-wrapper incoming-digital-task-title"
        id="tooltip-trigger-test-interaction-123"
      >
        customer@example.com
      </mdc-text>
      <mdc-tooltip
        class="task-tooltip"
        tooltip-type="description"
      >
        customer@example.com
      </mdc-tooltip>
      <mdc-text
        class="md-text-wrapper task-text"
      >
        Email Support
      </mdc-text>
      <mdc-text
        class="md-text-wrapper task-text"
      >
        Handle Time: 
          
        <div
          data-testid="mock-timer"
        >
          Timer
        </div>
      </mdc-text>
    </section>
  </div>
  <div
    data-position="end"
  >
    <div
      class="task-button-container"
    >
      <button
        class="md-button-pill-wrapper md-button-simple-wrapper"
        data-color="join"
        data-disabled="false"
        data-disabled-outline="false"
        data-ghost="false"
        data-grown="false"
        data-inverted="false"
        data-outline="false"
        data-shallow-disabled="false"
        data-size="40"
        data-testid="task:accept-button"
        tabindex="-1"
        type="button"
      >
        <span>
          Accept
        </span>
      </button>
    </div>
  </div>
</li>
`;

exports[`IncomingTaskComponent renders empty component when incomingTask is null 1`] = `null`;

exports[`IncomingTaskComponent renders incoming call and handles accept/decline for browser telephony task 1`] = `
<li
  class="task-list-item  task-list-hover md-list-item-base-wrapper"
  data-allow-text-select="false"
  data-disabled="false"
  data-interactive="true"
  data-padded="false"
  data-shape="rectangle"
  data-size="40"
  id="test-interaction-123"
  role="listitem"
  tabindex="0"
>
  <div
    data-position="start"
  >
    <mdc-avatar
      class="telephony"
      icon-name="handset-filled"
    />
  </div>
  <div
    data-position="fill"
  >
    <section
      class="task-details"
    >
      <mdc-text
        class="md-text-wrapper task-title"
      >
        1234567890
      </mdc-text>
      <mdc-text
        class="md-text-wrapper task-text"
      >
        Support Team
      </mdc-text>
      <mdc-text
        class="md-text-wrapper task-text"
      >
        Time Left: 
          
        <div
          data-testid="mock-timer"
        >
          Timer
        </div>
      </mdc-text>
    </section>
  </div>
  <div
    data-position="end"
  >
    <div
      class="task-button-container"
    >
      <button
        class="md-button-pill-wrapper md-button-simple-wrapper"
        data-color="join"
        data-disabled="false"
        data-disabled-outline="false"
        data-ghost="false"
        data-grown="false"
        data-inverted="false"
        data-outline="false"
        data-shallow-disabled="false"
        data-size="40"
        data-testid="task:accept-button"
        tabindex="-1"
        type="button"
      >
        <span>
          Accept
        </span>
      </button>
      <button
        class="md-button-pill-wrapper md-button-simple-wrapper"
        data-color="cancel"
        data-disabled="false"
        data-disabled-outline="false"
        data-ghost="false"
        data-grown="false"
        data-inverted="false"
        data-outline="false"
        data-shallow-disabled="false"
        data-size="40"
        data-testid="task:decline-button"
        tabindex="-1"
        type="button"
      >
        <span>
          Decline
        </span>
      </button>
    </div>
  </div>
</li>
`;

exports[`IncomingTaskComponent renders social media task with wrap up required 1`] = `
<li
  class="task-list-item  task-list-hover md-list-item-base-wrapper"
  data-allow-text-select="false"
  data-disabled="false"
  data-interactive="true"
  data-padded="false"
  data-shape="rectangle"
  data-size="40"
  id="test-interaction-123"
  role="listitem"
  tabindex="0"
>
  <div
    data-position="start"
  >
    <mdc-avatar
      class="social"
      icon-name="chat-filled"
    />
  </div>
  <div
    data-position="fill"
  >
    <section
      class="task-details"
    >
      <mdc-text
        class="md-text-wrapper incoming-digital-task-title"
        id="tooltip-trigger-test-interaction-123"
      >
        Bob Wilson
      </mdc-text>
      <mdc-tooltip
        class="task-tooltip"
        tooltip-type="description"
      >
        Bob Wilson
      </mdc-tooltip>
      <mdc-text
        class="md-text-wrapper task-text"
      >
        Social Support
      </mdc-text>
      <mdc-text
        class="md-text-wrapper task-text"
      >
        Handle Time: 
          
        <div
          data-testid="mock-timer"
        >
          Timer
        </div>
      </mdc-text>
    </section>
  </div>
  <div
    data-position="end"
  >
    <div
      class="task-button-container"
    />
  </div>
</li>
`;

exports[`IncomingTaskComponent renders task with missing call association details 1`] = `
<li
  class="task-list-item  task-list-hover md-list-item-base-wrapper"
  data-allow-text-select="false"
  data-disabled="false"
  data-interactive="true"
  data-padded="false"
  data-shape="rectangle"
  data-size="40"
  id="test-interaction-123"
  role="listitem"
  tabindex="0"
>
  <div
    data-position="start"
  >
    <mdc-avatar
      class="telephony"
      icon-name="handset-filled"
    />
  </div>
  <div
    data-position="fill"
  >
    <section
      class="task-details"
    >
      <mdc-text
        class="md-text-wrapper task-text"
      >
        Handle Time: 
          
        <div
          data-testid="mock-timer"
        >
          Timer
        </div>
      </mdc-text>
    </section>
  </div>
  <div
    data-position="end"
  >
    <div
      class="task-button-container"
    >
      <button
        class="md-button-pill-wrapper md-button-simple-wrapper"
        data-color="join"
        data-disabled="false"
        data-disabled-outline="false"
        data-ghost="false"
        data-grown="false"
        data-inverted="false"
        data-outline="false"
        data-shallow-disabled="false"
        data-size="40"
        data-testid="task:accept-button"
        tabindex="-1"
        type="button"
      >
        <span>
          Accept
        </span>
      </button>
      <button
        class="md-button-pill-wrapper md-button-simple-wrapper"
        data-color="cancel"
        data-disabled="false"
        data-disabled-outline="false"
        data-ghost="false"
        data-grown="false"
        data-inverted="false"
        data-outline="false"
        data-shallow-disabled="false"
        data-size="40"
        data-testid="task:decline-button"
        tabindex="-1"
        type="button"
      >
        <span>
          Decline
        </span>
      </button>
    </div>
  </div>
</li>
`;

exports[`IncomingTaskComponent renders task with special characters in ANI 1`] = `
<li
  class="task-list-item  task-list-hover md-list-item-base-wrapper"
  data-allow-text-select="false"
  data-disabled="false"
  data-interactive="true"
  data-padded="false"
  data-shape="rectangle"
  data-size="40"
  id="test-interaction-123"
  role="listitem"
  tabindex="0"
>
  <div
    data-position="start"
  >
    <mdc-avatar
      class="telephony"
      icon-name="handset-filled"
    />
  </div>
  <div
    data-position="fill"
  >
    <section
      class="task-details"
    >
      <mdc-text
        class="md-text-wrapper task-title"
      >
        +1 (555) 123-4567 ext. 123
      </mdc-text>
      <mdc-text
        class="md-text-wrapper task-text"
      >
        Special Support & Services
      </mdc-text>
      <mdc-text
        class="md-text-wrapper task-text"
      >
        Handle Time: 
          
        <div
          data-testid="mock-timer"
        >
          Timer
        </div>
      </mdc-text>
    </section>
  </div>
  <div
    data-position="end"
  >
    <div
      class="task-button-container"
    >
      <button
        class="md-button-pill-wrapper md-button-simple-wrapper"
        data-color="join"
        data-disabled="false"
        data-disabled-outline="false"
        data-ghost="false"
        data-grown="false"
        data-inverted="false"
        data-outline="false"
        data-shallow-disabled="false"
        data-size="40"
        data-testid="task:accept-button"
        tabindex="-1"
        type="button"
      >
        <span>
          Accept
        </span>
      </button>
      <button
        class="md-button-pill-wrapper md-button-simple-wrapper"
        data-color="cancel"
        data-disabled="false"
        data-disabled-outline="false"
        data-ghost="false"
        data-grown="false"
        data-inverted="false"
        data-outline="false"
        data-shallow-disabled="false"
        data-size="40"
        data-testid="task:decline-button"
        tabindex="-1"
        type="button"
      >
        <span>
          Decline
        </span>
      </button>
    </div>
  </div>
</li>
`;

exports[`IncomingTaskComponent shows correct button text for non-browser telephony task 1`] = `
<li
  class="task-list-item  task-list-hover md-list-item-base-wrapper"
  data-allow-text-select="false"
  data-disabled="false"
  data-interactive="true"
  data-padded="false"
  data-shape="rectangle"
  data-size="40"
  id="test-interaction-123"
  role="listitem"
  tabindex="0"
>
  <div
    data-position="start"
  >
    <mdc-avatar
      class="telephony"
      icon-name="handset-filled"
    />
  </div>
  <div
    data-position="fill"
  >
    <section
      class="task-details"
    >
      <mdc-text
        class="md-text-wrapper task-title"
      >
        1234567890
      </mdc-text>
      <mdc-text
        class="md-text-wrapper task-text"
      >
        Support Team
      </mdc-text>
      <mdc-text
        class="md-text-wrapper task-text"
      >
        Time Left: 
          
        <div
          data-testid="mock-timer"
        >
          Timer
        </div>
      </mdc-text>
    </section>
  </div>
  <div
    data-position="end"
  >
    <div
      class="task-button-container"
    >
      <button
        class="md-button-pill-wrapper md-button-simple-wrapper"
        data-color="join"
        data-disabled="true"
        data-disabled-outline="false"
        data-ghost="false"
        data-grown="false"
        data-inverted="false"
        data-outline="false"
        data-shallow-disabled="false"
        data-size="40"
        data-testid="task:accept-button"
        disabled=""
        type="button"
      >
        <span>
          Ringing...
        </span>
      </button>
    </div>
  </div>
</li>
`;
